x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { reactToUiComponent } from '../../../../../../../src/plugins/kibana_react/public';
+import { DASHBOARD_APP_URL_GENERATOR } from '../../../../../../../src/plugins/dashboard/public';
+import { PlaceContext, ActionContext, Config } from './types';
+import { CollectConfigContainer } from './components';
+import { DASHBOARD_TO_DASHBOARD_DRILLDOWN } from './constants';
+import { UiActionsEnhancedDrilldownDefinition as Drilldown } from '../../../../../advanced_ui_actions/public';
+import { txtGoToDashboard } from './i18n';
+import { esFilters } from '../../../../../../../src/plugins/data/public';
+import { VisualizeEmbeddableContract } from '../../../../../../../src/plugins/visualizations/public';
+import {
+  isRangeSelectTriggerContext,
+  isValueClickTriggerContext,
+} from '../../../../../../../src/plugins/embeddable/public';
+import { StartServicesGetter } from '../../../../../../../src/plugins/kibana_utils/public';
+import { StartDependencies } from '../../../plugin';
+
+export interface Params {
+  start: StartServicesGetter<Pick<StartDependencies, 'data' | 'advancedUiActions' | 'share'>>;
+}
+
+export class DashboardToDashboardDrilldown
+  implements Drilldown<Config, PlaceContext, ActionContext<VisualizeEmbeddableContract>> {
```

Comment:
Using `VisualizeEmbeddableContract` here I think is wrong, it should be able to work on any embeddable.

Another comment on this, does this have to be `ActionContext<VisualizeEmbeddableContract>`?  Why not just `EmbeddableContext`? 

I see:
`ActionContext<A> = A extends ActionDefinition<infer Context> ? Context : never;`
... but I can't really follow that. `VisualizeEmbeddableContract` isn't an ActionDefinition... sooo this is never?

Reply:
I guess, we should be able to generalize it in the future to work with more embeddable types. But right now, it looks like that `VisualizeEmbeddable` is the right type as the `execute()` function of this drilldown depends on the "input" of that embeddable type.

![image](https://user-images.githubusercontent.com/9773803/80693265-0849ac80-8ad3-11ea-88c5-89422ee9623c.png)

We can probably do some work-arounds around that usage of visualize embeddable "input", but I would suggest we do that in another PR as if we wanted this drilldown to be general, ideally we would have done it in [the PR where this drilldown was added to the feature branch](https://github.com/elastic/kibana/pull/63108).

Reply:
> Gotcha. The problem is that it is generic right now, but the type is not generic.

I don't understand what you mean.

> I can manage drilldowns for Lens: [...] I think we should either explicitly say "this only works right now for VisualizeEmbeddable" ...

That is expected, the dashboard drilldowns should work with any dashboard embeddable. They also should work with Lens; but right now Lens does not provide `.embeddable` field on `VALUE_CLICK_TRIGGER` context they execute; that should be easily to fix in Lens. But for now I've disable drilldowns for Lens embeddable.